cloudinit/sources/DataSourceAzure.py

```diff
@@ -559,6 +559,29 @@ def _get_data(self):
     def device_name_to_device(self, name):
         return self.ds_cfg['disk_aliases'].get(name)
 
+    @azure_ds_telemetry_reporter
+    def get_public_ssh_keys(self):
+        LOG.debug('Retrieving public SSH keys')
+        ssh_keys = []
+        try:
+            ssh_keys = [
+                public_key['keyData']
+                for public_key
+                in self.metadata['imds']['compute']['publicKeys']
+            ]
+            LOG.debug('Retrieved SSH keys from IMDS')
+        except KeyError:
+            LOG.debug('Unable to get keys from IMDS, falling back to OVF')
```

Comment:
msg = 'Unable to get keys from IMDS, falling back to OVF'
LOG.debug(msg)
report_diagnostic_event(msg)